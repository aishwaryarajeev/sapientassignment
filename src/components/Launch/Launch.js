import React from "react";
import { makeStyles } from "@material-ui/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import * as CONSTANTS from "../../constants";

const useStyles = makeStyles({
  root: {
    margin: "5px",
    width: "210px",
    height: "auto",
    "@media (min-width: 701px) and (max-width: 1024px) ": {
      flex: "0 0 33%",
      margin: "2% 5%",
    },
    "@media (min-width: 1025px) and (max-width: 1440px) ": {
      flex: "0 0 20%",
      margin: "2%",
    },
  },
  img: {
    backgroundColor: "#e6e6e6",
  },
  missionID: {
    fontWeight: 600,
    fontSize: "14px",
  },
  values: {
    margin: 0,
    fontSize: "13px",
    color: "#002b80",
    padding: "0px 8px",
  },
  heading: {
    marginBottom: "8px",
    fontSize: "12px",
    fontWeight: 600,
    color: "#002b80",
  },
  listItems: {
    marginLeft: "25px",
  },
});

export const Launch = ({ mission }) => {
  const {
    flight_number,
    mission_name,
    launch_year,
    launch_success,
    mission_id,
    links,
    rocket,
  } = mission;
  let { land_success } = rocket.first_stage.cores[0];
  const classes = useStyles();
  const renderValues = (title) => {
    switch (title) {
      case CONSTANTS.CARD_TITLES[1]:
        return launch_year;
      case CONSTANTS.CARD_TITLES[2]:
        return launch_success !== null ? launch_success.toString() : "NA";
      case CONSTANTS.CARD_TITLES[3]:
        return land_success !== null ? land_success.toString() : "NA";
      default:
        return;
    }
  };
  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography className={classes.img}>
          <img src={links.mission_patch_small} alt="mission" width="100%" />
        </Typography>
        <Typography className={classes.heading}>
          {mission_name}#{flight_number}
        </Typography>
        <Typography>
          {CONSTANTS.CARD_TITLES.map((title, index) => {
            return (
              <div>
                <span key={index} className={classes.missionID}>
                  {title}:
                </span>
                {index === 0 ? (
                  <div>
                    {mission_id.length > 0 ? (
                      mission_id.map((id) => {
                        return (
                          <ul key={id} className={classes.values}>
                            <li className={classes.listItems} key={id}>
                              {id}
                            </li>
                          </ul>
                        );
                      })
                    ) : (
                      <span className={classes.values}>
                        No mission id found
                      </span>
                    )}
                  </div>
                ) : (
                  <span className={classes.values}>{renderValues(title)}</span>
                )}
              </div>
            );
          })}
        </Typography>{" "}
      </CardContent>
    </Card>
  );
};
