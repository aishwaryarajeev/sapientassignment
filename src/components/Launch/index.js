export { Launch as default } from "./Launch";
