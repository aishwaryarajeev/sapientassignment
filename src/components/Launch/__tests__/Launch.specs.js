import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Launch } from "../Launch";

describe("<HomePage />", () => {
  let wrapper = null;
  const data = {
    flight_number: "1",
    launch_success: true,
    launch_year: "2009",
    links: {
      mission_patch_small: '"https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"',
    },
    mission_name: "FalconSat",
    mission_id: ["EE12345", "EE12345"],
    rocket: {
      first_stage: {
        cores: [
          {
            land_success: true,
          },
        ],
      },
    },
  };

  beforeEach(() => {
    wrapper = shallow(<Launch mission={data} />);
  });
  it("should render correctly", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
