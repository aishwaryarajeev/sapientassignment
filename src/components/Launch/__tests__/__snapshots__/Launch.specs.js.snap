// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render correctly 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-img-2"
    >
      <img
        alt="mission"
        src="\\"https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png\\""
        width="100%"
      />
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-heading-5"
    >
      FalconSat
      #
      1
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
      <div>
        <span
          className="makeStyles-missionID-3"
          key="0"
        >
          Mission Ids
          :
        </span>
        <div>
          <ul
            className="makeStyles-values-4"
            key="EE12345"
          >
            <li
              className="makeStyles-listItems-6"
              key="EE12345"
            >
              EE12345
            </li>
          </ul>
          <ul
            className="makeStyles-values-4"
            key="EE12345"
          >
            <li
              className="makeStyles-listItems-6"
              key="EE12345"
            >
              EE12345
            </li>
          </ul>
        </div>
      </div>
      <div>
        <span
          className="makeStyles-missionID-3"
          key="1"
        >
          Launch Year
          :
        </span>
        <span
          className="makeStyles-values-4"
        >
          2009
        </span>
      </div>
      <div>
        <span
          className="makeStyles-missionID-3"
          key="2"
        >
          Successful Launch
          :
        </span>
        <span
          className="makeStyles-values-4"
        >
          true
        </span>
      </div>
      <div>
        <span
          className="makeStyles-missionID-3"
          key="3"
        >
          Successful Landing
          :
        </span>
        <span
          className="makeStyles-values-4"
        >
          true
        </span>
      </div>
    </WithStyles(ForwardRef(Typography))>
     
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
